python/ray/serve/_private/router.py

```diff
@@ -202,10 +204,12 @@ def update_deployment_config(
                     ),
                 )
                 # Push metrics to the controller periodically.
-                self.metrics_pusher.register_or_update_task(
-                    self.PUSH_METRICS_TO_CONTROLLER_TASK_NAME,
-                    self.push_autoscaling_metrics_to_controller,
-                    autoscaling_config.metrics_interval_s,
+                shared = SharedHandleMetricsPusher.get_or_create(
+                    self._controller_handle
+                )
+                shared.register(self)
+                logger.info(
+                    f"Registered {self._handle_id} with shared metrics pusher {shared}."
                 )
             else:
```

Comment:
I'm not sure I understand this `else` block - it seems like even if `RAY_SERVE_COLLECT_AUTOSCALING_METRICS_ON_HANDLE = False` we still register the pushing task? It just won't have any data because the `_add_autoscaling_metrics_point` task isn't registered? 🤔